### ISSUE TITLE
bails out on non-integer data in timestamp

### ISSUE THREAD

**Issue by benfleis** [andrewning/sortphotos#54]
```
> sortphotos.py -t -r --sort '%Y/%m-%d' ./beka.jpg  ~/Pictures/
Preprocessing with ExifTool.  May take a while for a large number of files.
[{
  "SourceFile": "./beka.jpg",
  "File:FileModifyDate": "2005:02:04 06:23:11+01:00",
  "File:FileAccessDate": "2015:06:26 15:48:30+02:00",
  "File:FileInodeChangeDate": "2015:06:26 15:46:30+02:00",
  "EXIF:ModifyDate": "2004:12:03 08:36:27",
  "EXIF:DateTimeOriginal": "2004:12:01 16:26:55",
  "EXIF:CreateDate": "2004:12:01 16:26:55",
  "MakerNotes:YearCreated": 2004,
  "MakerNotes:MonthDayCreated": "12:01",
  "MakerNotes:TimeCreated": "16:26:55.04",
  "Composite:DateCreated": "2004:12:01"
}]
{ready}

Traceback (most recent call last):
  File "/Users/ben/bin/sortphotos.py", line 476, in <module>
    main()
  File "/Users/ben/bin/sortphotos.py", line 473, in main
    args.use_only_tags, not args.silent)
  File "/Users/ben/bin/sortphotos.py", line 305, in sortPhotos
    src_file, date, keys = get_oldest_timestamp(data, additional_groups_to_ignore, additional_tags_to_ignore)
  File "/Users/ben/bin/sortphotos.py", line 140, in get_oldest_timestamp
    exifdate = parse_date_exif(date)
  File "/Users/ben/bin/sortphotos.py", line 48, in parse_date_exif
    day = int(date_entries[2])
ValueError: invalid literal for int() with base 10: '55.04'
```

The failure is coming from: `"MakerNotes:TimeCreated": "16:26:55.04",`
Simple fix is to change line 48 to float, and then recast to int, but not sure what other impact that might have. I'm 100% naive here, but it seems harmless. If it sounds good, I'll fork and submit a PR for it.

**Comment by andrewning** [comment]
Yes do that. I've seen that error a couple times. I thought it was restricted to GPS time stamps but apparently not.

**Comment by andrewning** [comment]
Ok. I should have looked at this closer. It's not a day so this solution won't work. It's providing year, month/day, and time separately. I'm going to have to provide a different solution. For simplicity probably rejecting dates of this type (they should be redundant anyway).

**Comment by andrewning** [comment]
Fixed now with 08338fb
